lib/Platform/Intl/Constants.h

```diff
@@ -0,0 +1,322 @@
+/**
```

Comment:
I see a few problem with this file.

1. It is a header file but it contains definitions. Including it in more than one .cpp file will cause link errors.

2. It uses a lot of std::u16string for what appear to be constant strings. This will cause all of these to be heap allocated. Is that necessary?

3. All of the global constructors will cause startup delays. As a rule we try to avoid global constructors at all cost since they slow down initialization unconditionally. This file adds many many global constructirs.

Reply:
Thank you for the feedback, Tzvetan! I will work on updating the code to address the concerns raised.

I am thinking to replace all the struct variables here with namespaces and use ```inline constexpr auto&``` for all constant strings, per [this explanation](https://www.learncpp.com/cpp-tutorial/sharing-global-constants-across-multiple-files-using-inline-variables/). For example, change
```c++
const struct {
...
  const struct {
     std::u16string lookup_;
     std::u16string best_fit_;
  } matcher = {u"lookup", u"best fit"};
...
} optValue;
```
to
```c++
namespace opt_value {
...
namespace matcher {
inline constexpr auto& lookup = u"lookup";
inline constexpr auto& bestFit = u"best fit";
...
} // namespace matcher
} // namespace opt_value
```
And for the unordered_sets, I can replace them with arrays and specify inline constexpr. I believe that should address the concerns?

Any other suggestions are welcomed.

Reply:
You can follow the style used in `PlatformIntlApple.mm`, which is deliberately set up to avoid global constructors, and keep the strings close to the point where they are used (to make it easier to read).

Reply:
Instead of
```
inline constexpr auto& lookup = u"lookup";
```
can we just do:
```
inline constexpr char16_t lookup[] = u"lookup";
```

Reply:
@tmikov, yes, I will change to declare the type explicitly, rather than `auto`, which also yields char_16t[], per your guidance.

@neildhar, having the Intl public APIs' string constants in a separate Constant.h make them easier to review and check that they match the spec as a group. Another benefit is that the string constants that are semantically the same across the APIs like "localeMatcher" are then also shared across API implementation files. With the above suggested changes, there won't be global constructors involved. Your thoughts? Okay to go with this Constants.h grouping?

Reply:
Right. I am changing the `unordered_set` to array.

I am deleting the `unordered_map` usage in this file. It is for the Intl.DurationFormat API, which is currently a stage 3 TC39 proposal and not part of this PR. (I can put the map as a function local static inside that API implementation to avoid global constructor.)

I plan to push a new commit to update this PR next week to address the feedback so far 😀

Reply:
Updated PR with new commit to make changes as mentioned above.